contracts/soroban/contracts/xcall/src/states.rs

```diff
@@ -0,0 +1,248 @@
+use soroban_sdk::{Address, BytesN, Env, String, Vec};
+
+use crate::{
+    contract::Xcall,
+    errors::ContractError,
+    types::{
+        network_address::NetworkAddress,
+        request::CSMessageRequest,
+        rollback::Rollback,
+        storage_types::{Config, StorageKey},
+    },
+};
+
+pub const MAX_ROLLBACK_SIZE: u64 = 1024;
+pub const MAX_DATA_SIZE: u64 = 2048;
+
+impl Xcall {
+    pub fn is_initialized(e: &Env) -> Result<(), ContractError> {
+        let initialized = e.storage().instance().has(&StorageKey::Admin);
+        if initialized {
+            Err(ContractError::AlreadyInitialized)
+        } else {
+            Ok(())
+        }
+    }
+    pub fn admin(e: &Env) -> Result<Address, ContractError> {
+        if let Some(admin) = e.storage().instance().get(&StorageKey::Admin) {
+            Ok(admin)
+        } else {
+            Err(ContractError::Uninitialized)
+        }
+    }
+
+    pub fn get_config(e: &Env) -> Result<Config, ContractError> {
+        if let Some(config) = e.storage().instance().get(&StorageKey::Config) {
+            Ok(config)
+        } else {
+            Err(ContractError::Uninitialized)
+        }
```

Comment:
instead of this big blocks we can just do 
`e.storage().instance().get(&StorageKey::Config).ok_or(ContractError::Unintialized)`

Reply:
noted.

i will make changes accordingly for all block like this